packages/venia-concept/src/components/ProductFullDetail/ProductFullDetail.js

```diff
@@ -1,254 +1,241 @@
+};
 
-        if (!isConfigurable) {
-            return null;
-        }
-
-        return (
-            <Suspense fallback={fallback}>
-                <Options
-                    options={configurable_options}
-                    onSelectionChange={handleSelectionChange}
-                />
-            </Suspense>
-        );
+const getIsMissingOptions = (product, optionSelections) => {
+    // Non-configurable products can't be missing options.
+    if (!isProductConfigurable(product)) {
+        return false;
     }
 
-    get mediaGalleryEntries() {
-        const { props, state } = this;
-        const { product } = props;
-        const { optionCodes, optionSelections } = state;
-        const { media_gallery_entries, variants } = product;
+    // Configurable products are missing options if we have fewer
+    // option selections than the product has options.
+    const { configurable_options } = product;
+    const numProductOptions = configurable_options.length;
+    const numProductSelections = optionSelections.size;
 
-        const isConfigurable = isProductConfigurable(product);
+    return numProductSelections < numProductOptions;
+};
 
-        if (
-            !isConfigurable ||
-            (isConfigurable && optionSelections.size === 0)
-        ) {
-            return media_gallery_entries;
-        }
+const getMediaGalleryEntries = (product, optionCodes, optionSelections) => {
+    let value = [];
+
+    const { media_gallery_entries, variants } = product;
+    const isConfigurable = isProductConfigurable(product);
+    const optionsSelected = optionSelections.size > 0;
 
+    if (!isConfigurable || !optionsSelected) {
+        value = media_gallery_entries;
+    } else {
         const item = findMatchingVariant({
             optionCodes,
             optionSelections,
             variants
         });
 
-        if (!item) {
-            return media_gallery_entries;
-        }
-
-        return [
-            ...item.product.media_gallery_entries,
-            ...media_gallery_entries
-        ];
+        value = item
+            ? [...item.product.media_gallery_entries, ...media_gallery_entries]
+            : media_gallery_entries;
     }
 
-    get isMissingOptions() {
-        const { product } = this.props;
+    const key = value.reduce((fullKey, entry) => {
+        return `${fullKey},${entry.file}`;
+    }, '');
 
-        // Non-configurable products can't be missing options
-        if (!isProductConfigurable(product)) {
-            return false;
-        }
+    return { key, value };
+};
 
-        // Configurable products are missing options if we have fewer
-        // option selections than the product has options.
-        const { configurable_options } = product;
-        const numProductOptions = configurable_options.length;
-        const numProductSelections = this.state.optionSelections.size;
+const getProductOptions = (product, optionSelections, setOptionSelections) => {
```

Comment:
Any chance we can refactor this into a `ProductOptions` component and simplify the props? I think that'd look cleaner now that it basically is a function component.

Reply:
Resolved by https://github.com/magento-research/pwa-studio/pull/1240/commits/130afe85aa585973b1670b6f26468b2f25402c86.